ksqldb-execution/src/main/java/io/confluent/ksql/schema/query/QuerySchemas.java

```diff
@@ -118,47 +121,55 @@ public synchronized void trackSerdeOp(
     return Collections.unmodifiableMap(loggerToSchemas);
   }
 
+  Map<String, Map<Boolean, Set<String>>> getTopicsToLoggers() {
+    return Collections.unmodifiableMap(topicsToLoggers);
+  }
+
   /**
-   * Builds complete SchemaInfo for a topic:
+   * Returns all different schemas and serde formats that the {@code topicName} is using. The
+   * {@link MultiSchemaInfo} contains a list of key/value schemas and serde formats that were
+   * tracked for the specified topic.
+   * </p>
+   * Key and/or value schemas may be empty if no schemas were detected.
```

Comment:
Do you mean that the formats may be empty? Or are there situations where the schema might be empty too?

Reply:
Both. The QTT can't track what the format and/or schema is because a NULL value inserted in the value schema side. The left/outer joins has an internal `put(value, null)` as the first value in some cases, so QTT fails because it does not know what the format is.